tests/test_base.py

```diff
@@ -248,6 +249,13 @@ def test_set_coll_conformation(self, set_spec: Spec):
 
 
 class TestDictSpecValidation:
+    @pytest.mark.parametrize(
+        "pred", [{"id": str, s.opt("id"): int}, {s.opt("id"): int, "id": str},]
+    )
+    def test_dict_spec_definition(self, pred):
```

Comment:
Can you add a comment on the fact that this is testing duplicate keys please? Or maybe just change the test name.